tests/rules/parsing_method.js

```diff
@@ -0,0 +1,49 @@
+/**
+ * @file Test for no-unsanitized rule
+ * @author Frederik Braun et al.
+ * @copyright 2015-2017 Mozilla Corporation. All rights reserved
+ */
+
+//------------------------------------------------------------------------------
+// Requirements
+//------------------------------------------------------------------------------
+
+const rule = require("../../lib/rules/parsing_method");
+const RuleTester = require("eslint").RuleTester;
+
+//------------------------------------------------------------------------------
+// Tests
+//------------------------------------------------------------------------------
+
+const eslintTester = new RuleTester();
+
+eslintTester.run("parsing_method", rule, {
+    // Examples of code that should not trigger the rule
+    // XXX this does not find z['innerHTML'] and the like.
+
+    valid: [
+        {
+            code: "var doc = Document.parseHTMLUnsafe('static string');",
+        },
+    ],
+
+    // Examples of code that should trigger the rule
+    invalid: [
+        /* XXX Do NOT change the error strings below without review from freddy:
+         * The strings are optimized for SEO and understandability.
+         * The developer can search for them and will find this MDN article:
+         *  https://developer.mozilla.org/en-US/Firefox_OS/Security/Security_Automation
+         */
+
+        {
+            code: "var doc = Document.parseHTMLUnsafe(badness);",
+            errors: [
```

Comment:
> is there a way to make it a warning by moving "parsing" stuff into its own specific check? What happens if people would upgrade to a new linter version that includes "parsing_method"? Would they get new breakage? Would they just silently not get the new rule because of their settings?
> 
> If this breaks people, I'd be slightly more inclined to move the whole parsing_method into a completely different repo (that I will happily support)...

@mozfreddyb This line in the module declaring the plugin rules https://github.com/mozilla/eslint-plugin-no-unsanitized/pull/244/files#diff-e727e4bdf3657fd1d798edcd6b099d6e092f8573cba266154583a746bba0f346R22 should be the one setting the level of the issues identified by the parsing_method rule to warning instead of error, but I'll double-check by trying that locally to confirm that is actually the case.

Reply:
Oh I must have skipped that. No, that should totally work (and hopefully be configurable by users). Nice!